### PR TITLE
[WIP] Enable parallel for build:example

### LINKED ISSUE #2060: Parallelize the build:examples script
We now include the output Vega specs in the repo, using the `build:examples` (introduced in #2057).

However, `build:examples` is now single-thread.  We can make it faster using `parallel` as we use in `generate-images.sh`.  However, previously we assume that users of  `generate-images.sh` will manually install the right dependencies. 

Two things we have to guarantee:
- [x] Make sure `npm install` include the right dependency.
- [ ] Make sure travis includes the right dependency.

### PR DESCRIPTION
**Pull request by YuhanLu**
Part of #2060

### CODE REVIEW

**Comment by YuhanLu** [comment]
I still need to test if travis use parallel.

**Comment by YuhanLu** [comment]
I don't know why travis doesn't use parallel :(

**Comment by domoritz** [comment]
Maybe we need sudo. I think there is something in the travis docs.

File: scripts/build-examples.sh [hunk 1]
```diff
@@ -5,11 +5,17 @@ dir=${dir-"examples/vg-specs"}
...
+
+if type parallel >/dev/null 2>&1
+then
+  for file in examples/specs/*.vl.json; do
+    filename=$(basename "$file")
+    name="${filename%.vl.json}"
+    bin/vl2vg -p $file > $dir/$name.vg.json
+  done
+else
+  echo "yeah we use parallel!"
```
Comment (kanitw) [inline scripts/build-examples.sh]: echo "Using parallel to generate vega specs from examples in parallel"

For the other case, let's also say 


echo "Parallel not found! Sequentially generate vega specs from examples."

**Comment by kanitw** [comment]
- [ ] Probably should add a script to postinstall to make sure `parallel` is install for local development too?  (It's probably okay to make the script only work for mac and does not thing for Linux / window and add note to CONTRIBUTING.md.)

**Pull request merged**